Build failure under GCC 6+

```c++
cc -fPIC -O2 -g -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches   -m64 -mtune=generic -DDISCORD_PLUGIN_VERSION='"0.9.2017.05.25"' -shared -o libdiscord.so libdiscord.c -Wl,-z,relro  `pkg-config purple glib-2.0 json-glib-1.0 --libs --cflags`   -Ipurple2compat -g -ggdb
libdiscord.c: In function 'discord_get_user_flags':
libdiscord.c:916:2: error: 'for' loop initial declarations are only allowed in C99 mode
  for(guint i = 0; i < guild_membership->roles->len; i++){
  ^
libdiscord.c:916:2: note: use option -std=c99 or -std=gnu99 to compile your code
libdiscord.c: In function 'discord_process_dispatch':
libdiscord.c:1666:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (int j = json_array_get_length(members) - 1; j >= 0; j--) {
   ^
libdiscord.c:1676:4: error: 'for' loop initial declarations are only allowed in C99 mode
    for (int k = json_array_get_length(roles) - 1; k >= 0; k--) {
    ^
libdiscord.c:1685:12: error: redefinition of 'j'
   for (int j = json_array_get_length(presences) - 1; j >= 0; j--) {
            ^
libdiscord.c:1666:12: note: previous definition of 'j' was here
   for (int j = json_array_get_length(members) - 1; j >= 0; j--) {
            ^
libdiscord.c:1685:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (int j = json_array_get_length(presences) - 1; j >= 0; j--) {
   ^
libdiscord.c: In function 'discord_got_relationships':
libdiscord.c:2043:2: error: 'for' loop initial declarations are only allowed in C99 mode
  for (int i = len - 1; i >= 0; i--) {
  ^
libdiscord.c: In function 'discord_populate_guild':
libdiscord.c:2109:2: error: 'for' loop initial declarations are only allowed in C99 mode
  for (int j = json_array_get_length(roles) - 1; j >= 0; j--) {
  ^
libdiscord.c:2114:11: error: redefinition of 'j'
  for (int j = json_array_get_length(channels) - 1; j >= 0; j--) {
           ^
libdiscord.c:2109:11: note: previous definition of 'j' was here
  for (int j = json_array_get_length(roles) - 1; j >= 0; j--) {
           ^
libdiscord.c:2114:2: error: 'for' loop initial declarations are only allowed in C99 mode
  for (int j = json_array_get_length(channels) - 1; j >= 0; j--) {
  ^
libdiscord.c:2120:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for(int k = json_array_get_length(permission_overrides) - 1; k >= 0; k--){
   ^
libdiscord.c: In function 'discord_got_guilds':
libdiscord.c:2135:2: error: 'for' loop initial declarations are only allowed in C99 mode
  for (int i = len - 1; i >= 0; i--) {
  ^
libdiscord.c: In function 'discord_got_channel_info':
libdiscord.c:3043:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (int i = len - 1; i >= 0; i--) {
   ^
libdiscord.c:3056:14: error: conflicting types for 'i'
   for (guint i = 0; i < guild->members->len; i++){
              ^
libdiscord.c:3043:12: note: previous definition of 'i' was here
   for (int i = len - 1; i >= 0; i--) {
            ^
libdiscord.c:3056:3: error: 'for' loop initial declarations are only allowed in C99 mode
   for (guint i = 0; i < guild->members->len; i++){
   ^
make: *** [libdiscord.so] Error 1
```

--- a/Makefile
+++ b/Makefile
@@ -21,10 +21,10 @@ else
 PLUGIN_VERSION ?= 0.9.$(shell date +%Y.%m.%d)
 endif
 
-CFLAGS	?= -std=c99 -O2 -g -pipe -Wall
+CFLAGS	?= -O2 -g -pipe -Wall
 LDFLAGS ?= -Wl,-z,relro 
 
-CFLAGS  += -DDISCORD_PLUGIN_VERSION='"$(PLUGIN_VERSION)"'
+CFLAGS  += -std=c99 -DDISCORD_PLUGIN_VERSION='"$(PLUGIN_VERSION)"'
 
 # Do some nasty OS and purple version detection
 ifeq ($(OS),Windows_NT)

